### ISSUE TITLE
DataStore takes too much time to initialize

### ISSUE THREAD

**Issue by afern247** [aws-amplify/amplify-swift#1394]
### Describe the bug

As soon as I open the app, the user session remains (so let's discard that I'm checking for this for now).
As soon as the app starts, I query datastore for some data that the user has, this is the model:

```
type Wallet @model {
  id: ID!
  address: String!
  walletName: String
  chainsWithBalance: AWSJSON
  user: User @connection(name: "UserWallets")
}
```

so at app start I query like this:

```
func getUserWallets(user: AuthUser) async throws -> [Wallet] {
    try await withUnsafeThrowingContinuation { continuation in
        var allWallets: [Wallet] = []
        
        Amplify.DataStore.query(User.self, byId: user.userId) {
            switch $0 {
            case .success(let wallet):
                if let userWithWallets = wallet {
                    for wallet in userWithWallets.wallets! {
                        allWallets.append(wallet)
                    }
                } else {
                    print("⚠️ No wallets found for this user")
                }
                
                setWallet(wallet: allWallets)
                continuation.resume(returning: allWallets)
            case .failure(let error):
                DispatchQueue.main.async {
                    print("🛑 User not found - \(error.localizedDescription)")
                    continuation.resume(throwing: error)
                }
            }
        }
    }
}
```

The return of that is nil, it's just empty, however it does have some data. It is not after a short while that the data some up (me forcing it). Right now I have to use a counter counting from 4-5 seconds down and keep reloading the model quering for it until it's available, however the data is there... What's going on? Shouldn't it be available intimidatingly if it's offline? Also what's the best method to initialize this, like say: as soon as the data is up, receive an alert, like setting a var to true or something.

I just find this very frustrating, datastore should be able to come up in a second being an offline database instead of me looping through reload the data for 4 seconds.

### Amplify Categories

DataStore

### Dependency manager

Cocoapods

### Swift version

5.5

### CLI version

5.3.0

### Xcode version

13

### Device

iPhone 12

### iOS Version

15

**Comment by diegocstn** [comment]
@afern247 the DataStore full initial sync can take some time, especially with a large dataset. Which version of Amplify are you using? We've released a few performance improvements in[ v1.10.0.](https://github.com/aws-amplify/amplify-ios/releases/tag/v1.10.0)

> Also what's the best method to initialize this, like say: as soon as the data is up, receive an alert, like setting a var to true or something.

You can perhaps trigger a DataStore sync during your app startup time by using [`DataStore.start`](https://docs.amplify.aws/lib/datastore/other-methods/q/platform/ios/#start) and listen to one of the DataStore Hub [events](https://docs.amplify.aws/lib/datastore/datastore-events/q/platform/ios/#syncqueriesready) to get notified when data has been synced.

**Comment by afern247** [comment]
lol I'm using `1.12.0`. I don't even think my data is that much, it's actually veeery small, the most that could be there is the json but it's actually not that big.

The problem is that the data is there... like, it's ready to sync and I query for the data and it returns nil, then after 4 seconds do the same operation and it's there...
@diegocstn
